src/uwtools/tests/config/test_tools.py

```diff
@@ -667,3 +638,59 @@ def test__realize_config_values_needed_negative_results(caplog, tmp_path):
     assert "No keys are complete." in msgs
     assert "No keys have unrendered Jinja2 variables/expressions." in msgs
     assert "No keys are set to empty." in msgs
+
+
+@pytest.mark.parametrize("input_fmt", FORMATS)
+@pytest.mark.parametrize("output_fmt", FORMATS)
+def test__validate_format_output(input_fmt, output_fmt):
+    call = lambda: tools._validate_format_output(input_fmt=input_fmt, output_fmt=output_fmt)
+    if input_fmt in (FORMAT.yaml, output_fmt):
+        call()  # no exception raised
+    else:
+        with raises(UWError) as e:
+            call()
+        assert str(e.value) == f"Output format {output_fmt} must match input format {input_fmt}"
```

Comment:
Test all combinations of input and output formats. `call()` is expected to return without raising an exception if either the input format is YAML, in which case any output is fine; or if the input and output formats match. Otherwise, an exception is expected.